### PR TITLE
Support for Logitech G815 and G513

### PR DESCRIPTION
**Pull request by SamPoulton**
**List any issues that this PR fixes:** fixes # , etc...

Addresses #1778 
**This pull request proposes the following changes:**

- Support for Logitech G815 and G513

**Known issues/To do:**

- None

### CODE REVIEW

**Comment by diogotr7** [comment]
I noticed the g513 has an fn key, not a windows key. I also noticed the logitech integration contains no definition for an fn key? wondering if a cosmetic change would be sufficient

**Comment by SamPoulton** [comment]
I'll have a chat to the guy at #1778 tomorrow and see if he can try, I'll go from there

**Comment by diogotr7** [comment]
In any case you'd just have to add this to the `key_modifications` in the g513.json file

`    
"99": {
      "visualName": "FN",
      "tag": 99,
      "line_break": null,
      "margin_left": null,
      "margin_top": null,
      "width": null,
      "height": null,
      "font_size": null,
      "width_bits": null,
      "height_bits": null,
      "margin_left_bits": null,
      "margin_top_bits": null,
      "enabled": null
    }
`

**Comment by SamPoulton** [comment]
Is there any kind of official documentation for this kind of stuff anywhere?

**Comment by diogotr7** [comment]
No unfortunately. I know because I've added a few layouts before

**Comment by SamPoulton** [comment]
I've had no response from the person asking for G513 support, but I can confirm that the G815 works very nicely.

**Comment by PawelPleskaczynski** [comment]
Hi, do you know if this would work with my G815 with linear switches? I heard that there's a problem with this keyboard so G Hub can't recognize it. I can help to test this, but I'm not sure how to build this project on my computer, when I run "Build Solution" in Visual Studio 2019, I get these two errors:
```
Severity	Code	Description	Project	File	Line	Suppression State
Error	CS0006	Metadata file 'C:\Users\pawel\Projects\Aurora\Build\Release\Aurora.exe' could not be found	Aurora-Updater	C:\Users\pawel\Projects\Aurora\Project-Aurora\Aurora-Updater\CSC	1	N/A
```

```
Severity	Code	Description	Project	File	Line	Suppression State
Error		The tag 'KeySequence' does not exist in XML namespace 'clr-namespace:Aurora.Controls;assembly=Aurora'. Line 14 Position 10.	Plugin-Example	C:\Users\pawel\Projects\Aurora\Project-Aurora\Plugin-Example\Layers\Control_ExampleLayer.xaml	14	
```

**Comment by SamPoulton** [comment]
Just build Aurora itself rather than the entire solution

**Comment by PawelPleskaczynski** [comment]
Okay, I built it, but, as expected, it doesn't work with my G815 linear. I guess I'll have to return it, because it's faulty, no software can detect it

**Comment by SamPoulton** [comment]
@IhToN sorry but I accidentally closed the PR and the review was deleted, any chance you'd be able to do it again? Nothing's changed

**Comment by diogotr7** [comment]
Hey, sorry i'm taking so long to get through the pulls.

Is this functional? I don't know if the M keys on the g815 are supported by logitech's sdk.

File: Project-Aurora/Project-Aurora/Settings/Configuration.cs [hunk 1]
```diff
@@ -178,7 +178,11 @@ public enum PreferredKeyboard
         Logitech_GPRO = 103,
 		[Description("Logitech - G213")]
         Logitech_G213 = 104,
-
+		[Description("Logitech - G815")]
```
Comment (diogotr7) [inline Project-Aurora/Project-Aurora/Settings/Configuration.cs:181]: please fix the tabs

File: Project-Aurora/Project-Aurora/Settings/KeyboardLayoutManager.cs [hunk 2]
```diff
@@ -690,6 +690,12 @@ public void LoadBrand(PreferredKeyboard keyboard_preference = PreferredKeyboard.
                 layoutConfigPath = Path.Combine(layoutsPath, "logitech_gpro.json");
             else if (keyboard_preference == PreferredKeyboard.Logitech_G410)
                 layoutConfigPath = Path.Combine(layoutsPath, "logitech_g410.json");
+			      else if (keyboard_preference == PreferredKeyboard.Logitech_G213)
```
Comment (diogotr7) [inline Project-Aurora/Project-Aurora/Settings/KeyboardLayoutManager.cs:693]: the g213 is present twice

**Comment by IhToN** [comment]
> 
> 
> Hey, sorry i'm taking so long to get through the pulls.
> 
> Is this functional? I don't know if the M keys on the g815 are supported by logitech's sdk.

The M keys are not supported you are right. Although the G1-5 ones are missing on the layout for G810.

**Comment by diogotr7** [comment]
The g810 does not have any macro keys.

**Comment by IhToN** [comment]
> 
> 
> The g810 does not have any macro keys.

I know, I meant by using the G810 Layout as a temporary solution for the G815

**Comment by diogotr7** [comment]
I decided to fix the issues i pointed out myself, i hope that's okay.

Please test this if you don't mind as i dont have the keyboards to test with.

**Pull request merged**